compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmPackageGen.java

```diff
@@ -807,17 +812,32 @@ CompiledJarFile generate(BIRPackage module, boolean isEntry) {
         return new CompiledJarFile(getModuleLevelClassName(module.packageID, MODULE_INIT_CLASS_NAME, "."), jarEntries);
     }
 
-    private BIRNode.BIRFunction getMainFunc(List<BIRNode.BIRFunction> funcs) {
-        BIRNode.BIRFunction userMainFunc = null;
-        for (BIRNode.BIRFunction func : funcs) {
-            if (func != null && func.name.value.equals(MAIN_METHOD)) {
-                userMainFunc = func;
-                break;
+    private BIRFunction getMainFunc(BIRPackage module) {
+        BIRFunction userMainFunc = null;
+        if (module.packageID.skipTests) {
+            for (BIRFunction func : module.functions) {
+                if (func.name.value.equals(MAIN_METHOD)) {
+                    userMainFunc = func;
+                    break;
+                }
             }
         }
         return userMainFunc;
     }
 
+    private BIRFunction getTestExecuteFunc(BIRPackage module) {
```

Comment:
As we have same logic for `getMainFunc` function as well, we can reuse it by passing the function name.